libraries/AP_HAL_ChibiOS/hwdef/thepeach-k1/hwdef.dat

```diff
@@ -0,0 +1,292 @@
+PC0 VBUS_nVALID INPUT PULLUP
+
+# This defines more ADC inputs.
+PC3 AUX_POWER ADC1 SCALE(1)
+PC4 AUX_ADC2 ADC1 SCALE(1)
+
+# And the analog input for airspeed (rarely used these days).
+PC5 PRESSURE_SENS ADC1 SCALE(2)
+
+# This sets up the UART for talking to the IOMCU. Note that it is
+# vital that this UART has DMA available. See the DMA settings below
+# for more information.
+
+#  USART6 to IO
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+
+# Now setup the pins for the microSD card, if available.
+PC8 SDIO_D0 SDIO
+PC9 SDIO_D1 SDIO
+PC10 SDIO_D2 SDIO
+PC11 SDIO_D3 SDIO
+PC12 SDIO_CK SDIO
+PD2 SDIO_CMD SDIO
+
+# More CS pins for more sensors. The labels for all CS pins need to
+# match the SPI device table later in this file.
+PC2 MPU9250_CS CS
+PC15 ICM20602_CS CS
+PD7 MS5611_CS CS
+
+# the first CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+# Another USART, this one for telem1. This one has RTS and CTS lines.
+# USART2 serial2 telem1
+PD3 USART2_CTS USART2
+PD4 USART2_RTS USART2
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+
+# The telem2 USART, also with RTS/CTS available.
+# USART3 serial3 telem2
+PD8 USART3_TX USART3
+PD9 USART3_RX USART3
+PD11 USART3_CTS USART3
+PD12 USART3_RTS USART3
+
+# The CS pin for FRAM (ramtron). This one is marked as using
+# SPEED_VERYLOW, which matches the HAL_PX4 setup.
+PD10 FRAM_CS CS SPEED_VERYLOW
+
+# Now we start defining some PWM pins. We also map these pins to GPIO
+# values, so users can set BRD_PWM_COUNT to choose how many of the PWM
+# outputs on the primary MCU are setup as PWM and how many as
+# GPIOs. To match HAL_PX4 we number the GPIOs for the PWM outputs
+# starting at 50.
+PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
+PE13 TIM1_CH3 TIM1 PWM(2) GPIO(51)
+PE11 TIM1_CH2 TIM1 PWM(3) GPIO(52)
+PE9  TIM1_CH1 TIM1 PWM(4) GPIO(53)
+PD13 TIM4_CH2 TIM4 PWM(5) GPIO(54)
+#PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55)
+
+# This is the invensense data-ready pin. We don't use it in the
+# default driver.
+PC14 ICM20602_DRDY INPUT
+PD15 MPU9250_DRDY INPUT
+
+# the 2nd GPS UART
+# UART8 serial4 GPS2
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# This is the pin to enable the sensors rail. It can be used to power
+# cycle sensors to recover them in case there are problems with power on
+# timing affecting sensor stability. We pull it high by default.
+PE3 VDD_3V3_SENSORS_EN OUTPUT HIGH
+
+# UART7 maps to uartF in the HAL (serial5 in SERIALn_ parameters).
+PE7 UART7_RX UART7
+PE8 UART7_TX UART7
+
+# Define a LED, mapping it to GPIO(0). LOW will illuminate the LED
+PE12 FMU_LED_AMBER OUTPUT HIGH OPENDRAIN GPIO(0)
+
+# Power flag pins: these tell the MCU the status of the various power
+# supplies that are available. The pin names need to exactly match the
+# names used in AnalogIn.cpp. 
+PB5 VDD_BRICK_nVALID INPUT PULLUP
+PE15 VDD_5V_PERIPH_nOC INPUT PULLUP
+
+# SPI devices
+SPIDEV ms5611         SPI1 DEVID3  MS5611_CS    MODE3  20*MHZ 20*MHZ
+SPIDEV icm20602       SPI1 DEVID1  ICM20602_CS  MODE3   4*MHZ  8*MHZ
```

Comment:
I could be wrong, but I thought we tended to use lower speeds for SPI SLOW for reliability reasons

Reply:
that is only needed on some invensense sensors